internal/app/server/handlers.go

```diff
@@ -0,0 +1,94 @@
+package server
+
+import (
+	"fmt"
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/ikashurnikov/shortener/internal/app/shortener"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/go-chi/chi/v5"
+)
+
+type Handler struct {
+	*chi.Mux
+	shortener shortener.Shortener
+	baseURL   url.URL
+}
+
+func NewHandler(shortener shortener.Shortener, baseURL url.URL) *Handler {
+	badRequest := func(writer http.ResponseWriter, request *http.Request) {
+		writer.WriteHeader(http.StatusBadRequest)
+	}
+
+	router := chi.NewRouter()
+	router.Use(middleware.CleanPath)
+	router.Use(middleware.RequestID)
+	router.Use(middleware.RealIP)
+	router.Use(middleware.Recoverer)
+	router.Use(middleware.Logger)
+	router.NotFound(badRequest)
+	router.MethodNotAllowed(badRequest)
+
+	handler := &Handler{
+		Mux:       router,
+		shortener: shortener,
+		baseURL:   baseURL,
+	}
+	handler.Route("/", func(router chi.Router) {
+		router.Post("/", handler.postLongLink())
+		router.Get("/{shortURL}", handler.getShortLink())
+	})
+	return handler
+}
+
+func (handler *Handler) postLongLink() http.HandlerFunc {
+	return func(responseWriter http.ResponseWriter, request *http.Request) {
```

Comment:
можно просто 
```suggestion
func (handler *Handler) postLongLink(responseWriter http.ResponseWriter, request *http.Request) {
```
и тогда в ройтере можно так написать `router.Post("/", handler.postLongLink)`